arch/risc-v/src/common/riscv_mtimer.c

```diff
@@ -83,7 +83,7 @@ static const struct oneshot_operations_s g_riscv_mtimer_ops =
 #ifndef CONFIG_ARCH_USE_S_MODE
 static uint64_t riscv_mtimer_get_mtime(struct riscv_mtimer_lowerhalf_s *priv)
 {
-#ifdef CONFIG_ARCH_RV64
+#if defined(CONFIG_ARCH_RV64) && !defined(CONFIG_ARCH_CHIP_K230)
```

Comment:
Good catch! This is quick-n-dirty solution as I mentioned in maillist.

Other solutions I can think of are:

a) See if current RiscV layer design already allows overriding this common layer behavior at chip specific layer.
b) Refactoring the common layer so that to use explicit config options for this CLINT MMIO bus width choice. Current design assumption of "64-bit chips will do 64-bit CLINT MMIO" doesn't hold true for chips like k230. However, doing this will need broader consensus as it impacts all Risc-V ports.

As I am new with NuttX, I don't know if option a holds or not at this time. If it doesn't hold, then I would suggest to take current solution first. Then the common layaer have option b done, then we can clean up the dirty one.
 
Please teach your suggestions.

Reply:
Good catch! This is quick-n-dirty solution as I mentioned in maillist.

Other solutions I can think of are:

a) See if current RiscV layer design already allows overriding this common layer behavior at chip specific layer.
b) Refactoring the common layer so that to use explicit config options for this CLINT MMIO bus width choice. Current design assumption of "64-bit chips will do 64-bit CLINT MMIO" doesn't hold true for chips like k230. However, doing this will need broader consensus as it impacts all Risc-V ports.

As I am new with NuttX, I don't know if option a holds or not at this time. If it doesn't hold, then I would suggest to take current solution first. Then the common layaer have option b done, then we can clean up the dirty one.
 
Please teach your suggestions.

Reply:
I think we can add an option(e.g. ARCH_RV_MMIO_NO_64BIT)
```
config ARCH_RV_MMIO_NO_64BIT
	bool
	depnds on ARCH_RV64
```
and select ARCH_RV_MMIO_NO_64BIT under ARCH_RV64.

Reply:
@xiaoxiang781216

if you are sure that non-RISCV architectures won't have this different MMIO width issue, I am fine to contain the issue within RISCV scope for now. Otherwise later you will see that ARCH_MMIO_NO_64BIT is more proper.  

On the other side I am concerned with this NO_64BIT style design as when 128 bit RiscV chip arrives later you will need add MMIO_NO_128BIT.  

Maybe we can consider explicit options like::

- two booleans like `ARCH_RV_MMIO_64BIT` and `ARCH_RV_MMIO_32BIT`?
- one enum like `ARCH_RV_MMIO_BITS`?

Reply:
It's fine to put either the common place or risc-v specific, just provide the change you think it's best.

Reply:
I don't know NuttX well enough, and I need time to learn Kconfig tweaking if I must do it. In general it is a little rushy to include such change with this PR as it will impact other ports that I may not have chance to test. That is why my original suggestion is to take a debt for now and notice common design group to figure out a final solution.

Reply:
@xiaoxiang781216 

Suppose we introduce a new int type `ARCH_RV_MMIO_BITS` which defaults to 32 or 63 depending on ARCH_RV32 or ARCH_RV64 respectively. How we override its value to 32 for ARCH_CHIP_K230? I

Reply:
@xiaoxiang781216 I've update the PR using newly added `ARCH_RV_MMIO_BITS` config above. 
I did `ostest` checks with `rv-virt` and `canmv230`, plus manual config checks for a few RiscV ports.

Please review and let me know if there is anything else.